website/docs/timeline.md

```diff
@@ -3,40 +3,384 @@ title: Timeline
+- **INDEXING**: This denotes an action to [asynchronounsly build the index](https://hudi.apache.org/docs/metadata_indexing) in the metadata table.
+
+## States
+A state denotes if an action has been scheduled, in progress or completed. There are three different states an action can be in: 
+- **REQUESTED**: Denotes an action has been scheduled, but has not initiated.
+- **INFLIGHT**: Denotes an action is currently being performed.
+- **COMPLETED**: Denotes completion of an action on the timeline.
+
+When a transaction is initiated for an operation, we create an instant file with a `.requested` suffix that marks the transaction has been initiated. When a transaction is transitioned from requested to inflight, the writes or a table service task (i.e., cleaning, compaction, etc) will be modifying, deleting, updating or inserting a file in a Hudi table. The instant file will have a `.inflight`  suffix, and the inflight state means the transaction tasks are ongoing. When the transaction tasks are completed successfully, the transaction state transitions from inflight to complete. The instant file will **NOT HAVE** a completed suffix; instead, there will be no suffix. We’ll point this out down below. 
+
+## Instant
+At its core, Hudi maintains a timeline of all actions performed on the table at different timestamps, represented as a series of instants. The instants provide snapshot views of the table, while also efficiently supporting data retrieval in the order of arrival. A Hudi instant consists of the following components:
+- **instant time**: An instant time is typically a timestamp (e.g: 20190117010349 for second-level granularity prior to 0.10.0 release or 20190117010349567 for millisecond-level granularity since 0.10.0 release). The instant time monotonically increases in the order of action's begin time. The timezone will be the same as your machine's timeszone.
+- **instant action**:  An instant action is a type of action performed on the table. 
+- **State suffix**: A state suffix is the current state of the instant.
+
+Inside the `.hoodie` folder, each transaction has three metadata files. Each file is named differently depending on the state: 
+- For a transaction with a **requested** or **inflight** state, the files are written like this: 
+  - [instantTime.instantAction.stateSuffix]
+ - For a transaction with a **completed** state, the files are written like this: 
+   - [instantTime.ActionName]
+
+Below are four instances inside the `.hoodie` folder. I want to highlight two instant examples where the states are different:
+
+```sh
+cd .hoodie && ls 
+20230210181040140.commit.requested     #requested state
+2023021018095339.commit                #completed state
+20230210180953939.commit.requested  
+20230210180953939.inflight
+
+```
+
+- Looking at the second line where there is a **#requested** comment, we can see this instant file has the timestamp + action + state because it was in a requested state.
```

Comment:
These line numbers seems wrong?

Reply:
updated to refer to the exact instance:

- Looking at `20230210181040140.commit.requested `, we can see this instant file has the timestamp + action + state because it was in a requested state.

- Looking at `2023021018095339.commit`, we can see this instant has the timestamp + action because it is in the completed state.

Reply:
wdyt @danny0405 ?